aten/src/ATen/native/Repeat.cpp

```diff
@@ -97,31 +97,30 @@ Tensor repeat_interleave(
   return ret;
 }
 
-Tensor repeat_interleave(
-    const Tensor& self,
-    int64_t repeats,
-    c10::optional<int64_t> dim,
-    c10::optional<int64_t> output_size) {
-  Tensor input = self;
-  at::Tensor repeats_ = at::empty(1, self.options().dtype(at::kLong)).fill_(repeats);
-  if (!output_size) {
-    if (!dim) {
-      input = input.flatten();
-      dim = 0;
-    }
-    auto input_size = input.sym_size(dim.value()).guard_int(__FILE__, __LINE__);
-    output_size = input_size * repeats;
-  }
-  return at::native::repeat_interleave(input, repeats_, dim, output_size);
-}
-
 Tensor repeat_interleave_symint(
     const Tensor& self,
     c10::SymInt repeats,
-    c10::optional<int64_t> dim,
+    c10::optional<int64_t> dim_opt,
     c10::optional<int64_t> output_size) {
-    return at::native::repeat_interleave(self, repeats.guard_int(__FILE__, __LINE__), dim, output_size);
+  Tensor input = dim_opt ? self : self.flatten();
+  int64_t dim = c10::maybe_wrap_dim(dim_opt.value_or(0), self.dim());
+  TORCH_CHECK(repeats >= 0, "Repeats must be non-negative");
+
+  input = input.unsqueeze(dim + 1);
+  auto expand_shape = input.sym_sizes().vec();
+  expand_shape[dim + 1] = repeats;
+  input = input.expand_symint(expand_shape);
+
+  // This argument doesn't really make sense for the scalar overload, but exists
+  // for consistency with the tensor overload
+  if (output_size) {
+    auto calculated_size = (repeats * expand_shape[dim]).guard_int(__FILE__, __LINE__);
+    TORCH_CHECK(calculated_size == *output_size, "repeat_interleave: Invalid output_size, expected ",
+                calculated_size, " but got ", *output_size);
   }
 
+  return input.clone().flatten(dim, dim + 1);
```

Comment:
This is potentially copying twice. What about doing the flatten first, then check if we copied, and if we didn't then clone?

Reply:
I've switched it to force contiguous format, which should guarantee that `flatten` doesn't ever copy. I also checked and the old implementation returned contiguous as well,  so this should be fine.